### ISSUE TITLE
looks_like_git does not recognize GitLab's embedded git

### ISSUE THREAD

**Issue by gnustavo** [book/Git-Version-Compare#1]
I've created a [server hook](https://docs.gitlab.com/ee/administration/server_hooks.html) in a GitLab repository which uses Git::Version::Compare (via Git::Repository and Git::Hooks modules) but it fails because GitLab uses an "embedded" git which has a particular version string. The hook error message is this:

```
remote: 2.37.1.gl1 does not look like a Git version at (eval 22) line 5.
```

Note that non-standard `gl1` suffix in the version string.

I thought about a few different strategies to make Git::Version::Compare accept this.

The simplest and most specific way would be to change the regexp in the looks_like_git routine so that it accepts "gl" as an alternative to "rc":

```
-          (?:[.-]?rc[0-9]+)?                                  # rc
+          (?:[.-]?(?:rc|gl)[0-9]+)?                            # rc|gl
```

Or one could be more relaxed and accept any sequence of letters in the suffix:

```
-          (?:[.-]?rc[0-9]+)?                                  # rc
+          (?:[.-]?[a-z]+[0-9]+)?                               # rc, gl, etc.
```

An yet more generic option would be to allow one to change the regexp via a configuration variable. Either the whole regexp could be a global variable or just the suffix part. That way, one could specialize the regexp in a specific script.

What do you think about accepting such non-standard version strings?

**Comment by book** [comment]
Given that that portion of the version is ignored unless it starts with `rc`, I think the more generic version should be fine. It's very likely that other vendors will also come up with some fancy suffixes too, so we'd better be prepared for them.

If you send a patch, please add some real world example values to `t/looks_like_git.t`.

**Comment by gnustavo** [comment]
Great! I can make a patch to send as a PR. But it isn't clear to me exactly which option you prefer out of the three I suggested. Is it the second or the third one?

If it is the third option, would you prefer to allow the user to change the whole regexp or only the alphabetic part of the suffix (replacing the `rc` part)?

**Comment by book** [comment]
I don't think it's going to help much to allow the user to pass a regexp of their choice.

Just replacing the portion of the regexp that deals with `rc` should be enough to cover most cases of non-standard versions.

So:

    (?:[.-]?[a-z]+[0-9]+)?

**Comment by gnustavo** [comment]
Thanks. I just submitted PR #2 for your consideration.

**Comment by book** [comment]
Fixed by #2.
